### PR TITLE
feat(cache): prefer If-Modified-Since over If-None-Match

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Set Last-Modified header to http response cache if present and prefer it over If-None-Match (etag).

## Context

GitHub caching apparently works more reliably based on dates than etags.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
To be combined with #26793

File: lib/util/http/index.ts [hunk 1]
```diff
@@ -214,12 +214,23 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
...
-          logger.debug(`Using cached etag for ${url}`);
+        const responseCache = getCache().httpCache?.[url];
+        // Prefer If-Modified-Since over If-None-Match
+        if (responseCache?.['lastModified']) {
+          logger.debug(
+            `http cache: trying cached Last-Modified "${responseCache?.['lastModified']}" for ${url}`,
+          );
+          options.headers = {
+            ...options.headers,
+            'If-Modified-Since': responseCache['lastModified'],
```
Comment (secustor) [inline lib/util/http/index.ts:225]: Don't we wanna save when we have taken the last look at that URL?

Comment (rarkins) [reply lib/util/http/index.ts:225]: I think we do. This part is when we're crafting the cached request later

Comment (rarkins) [reply lib/util/http/index.ts:225]: See line 284

**Review by secustor — APPROVED** [review]

**Pull request merged**